contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -3,94 +3,20 @@
-        string calldata destinationContractAddress,
-        bytes calldata payload
-    ) external {
-        emit ContractCall(msg.sender, destinationChain, destinationContractAddress, keccak256(payload), payload);
-    }
-
-    function isContractCallApproved(
-        bytes32 commandId,
-        string calldata sourceChain,
-        string calldata sourceAddress,
-        address contractAddress,
-        bytes32 payloadHash
-    ) external view override returns (bool) {
-        return _isContractCallApproved(commandId, sourceChain, sourceAddress, contractAddress, payloadHash);
-    }
-
-    function isMessageApproved(
-        string calldata messageId,
-        string calldata sourceChain,
-        string calldata sourceAddress,
-        address contractAddress,
-        bytes32 payloadHash
-    ) external view override returns (bool) {
-        bytes32 commandId = messageToCommandId(sourceChain, messageId);
-        return _isContractCallApproved(commandId, sourceChain, sourceAddress, contractAddress, payloadHash);
-    }
 
-    function validateContractCall(
-        bytes32 commandId,
-        string calldata sourceChain,
-        string calldata sourceAddress,
-        bytes32 payloadHash
-    ) external override returns (bool valid) {
-        valid = _validateContractCall(commandId, sourceChain, sourceAddress, payloadHash);
-    }
+import { CommandType, Message } from '../types/AmplifierGatewayTypes.sol';
+import { WeightedSigners, Proof } from '../types/WeightedMultisigTypes.sol';
 
-    function validateMessage(
-        string calldata messageId,
-        string calldata sourceChain,
-        string calldata sourceAddress,
-        bytes32 payloadHash
-    ) external override returns (bool valid) {
-        bytes32 commandId = messageToCommandId(sourceChain, messageId);
-        valid = _validateContractCall(commandId, sourceChain, sourceAddress, payloadHash);
-    }
+import { BaseWeightedMultisig } from '../governance/BaseWeightedMultisig.sol';
+import { BaseAmplifierGateway } from './BaseAmplifierGateway.sol';
 
-    function isCommandExecuted(bytes32 commandId) public view override returns (bool) {
-        return _storage().commands[commandId];
-    }
-
-    /**
-     * @notice Compute the commandId for a `Message`.
-     * @param sourceChain The name of the source chain as registered on Axelar.
-     * @param messageId The unique message id for the message.
-     * @return The commandId for the message.
-     */
-    function messageToCommandId(string calldata sourceChain, string calldata messageId) public pure returns (bytes32) {
-        // Axelar prevents `sourceChain` to contain '_',
-        // hence we can use it as a separator with abi.encodePacked to avoid ambiguous encodings
-        return keccak256(abi.encodePacked(CommandType.ApproveMessages, sourceChain, '_', messageId));
+contract AxelarAmplifierGateway is BaseAmplifierGateway, BaseWeightedMultisig, IAxelarAmplifierGateway {
```

Comment:
Why are you not inheriting `AxelarAmplifierAuth` instead of `BaseWeightedMultisig`?

Reply:
If not maybe we can just delete `AxelarAmplifierAuth`?

Reply:
yeah, going to delete it in a separate PR, need to change tests more but then this PR would be harder to read

Reply:
Gotcha